assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -402,8 +517,58 @@ export const controls = {
 	),
 };
 
+// eslint-disable-next-line complexity
 export const reducer = createReducer( ( state, { type, payload } ) => {
 	switch ( type ) {
+		case INSERT_NOTIFICATION_INTO_RESOLVED_QUEUE: {
+			const { id } = payload;
+
+			/**
+			 * The notification we want to add to the already-resolved queue.
+			 */
+			const notification = state.notifications?.[ id ];
+
+			if ( notification === undefined ) {
+				global.console.warn(
+					`Could not add notification with ID "${ id }" to queue. Notification "${ id }" is not registered.`
+				);
+
+				break;
+			}
+
+			state.queuedNotifications[ notification.groupID ] =
+				state.queuedNotifications[ notification.groupID ] || [];
+
+			// If the notification is already in the queue, we don't need to add it
+			// again.
+			if (
+				state.queuedNotifications[ notification.groupID ].some(
+					( notificationInQueue ) => notificationInQueue.id === id
+				)
+			) {
+				break;
+			}
+
+			// Find the next notification in the queue that has a lower priority than
+			// the one we're adding, and add this notification ahead of it.
+			const positionForNewNotification = state.queuedNotifications[
+				notification.groupID
+			].findIndex( ( notificationInQueue ) => {
```

Comment:
I can never remember what the return of `findIndex` means. It would be useful to decode this above with the comment there which is also useful.

Reply:
Hah, totally, I'll do that 👍🏻